aws/resource_aws_api_gateway_documentation_version.go

```diff
@@ -0,0 +1,138 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayDocumentationVersion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayDocumentationVersionCreate,
+		Read:   resourceAwsApiGatewayDocumentationVersionRead,
+		Update: resourceAwsApiGatewayDocumentationVersionUpdate,
+		Delete: resourceAwsApiGatewayDocumentationVersionDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayDocumentationVersionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	restApiId := d.Get("rest_api_id").(string)
+
+	params := &apigateway.CreateDocumentationVersionInput{
```

Comment:
Question: the [`CreateDocumentationVersionInput` type](https://docs.aws.amazon.com/sdk-for-go/api/service/apigateway/#CreateDocumentationVersionInput) also supports `StageName`, do we need to support it as well?
```
// The stage name to be associated with the new documentation snapshot.
StageName *string `locationName:"stageName" type:"string"`
```

Reply:
I intentionally removed it, although it was in my original implementation. CloudFormation also doesn't have it, probably for the same reason https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-documentationversion.html

It has a very problematic workflow. You need to create the stage 1st (for referencing in APIG Documentation Version), but you cannot delete the APIG Documentation Version before deleting the stage. Basically the dependency chain reverts during destruction.

Either way we shouldn't need it, because https://www.terraform.io/docs/providers/aws/r/api_gateway_stage.html#documentation_version allows users to associate the two.

Reply:
👍  sounds good